해당 이슈는 PyMySQL 1.1.0v 에서 발생한 것으로, SQL 인젝션 취약점이다.

취약한 코드는 다음과 같다.
```python
from sqlalchemy import text


async def findById(session, id: str):  
    try:  
        with session.begin():  
            sql = text(f"SELECT * FROM user WHERE id = {id}")  
            result = session.execute(sql).fetchall()   
            return result
    except Exception as e:  
        session.rollback()  
        raise e  
    finally:  
        session.close()
```

안전한 코드
```python
from sqlalchemy import text


async def findById(session, id: str):  
    try:  
        with session.begin():  
            sql = text(f"SELECT * FROM user WHERE id = :id")  
            result = session.execute(sql, ({"id": id})).fetchall()   
            return result
    except Exception as e:  
        session.rollback()  
        raise e  
    finally:  
        session.close()
```

보기만해선 모르겠으니 직접 해보도록 하자.
id는 모두 "1 or 1=1"을 넣어준다. (or과 함께 1=1이므로 Where문은 모든 row에 대해 True이다)

취약한 코드일 때
```json
{
	"data": [
		{
			"id": 1,
			"name": "이름_a",
			"a_id": "name_1",
			"password": "1234"
		},
		{
			"id": 2,
			"name": "이름_b",
			"a_id": "name_2",
			"password": "1234"
		},
		{
			"id": 3,
			"name": "이름_c",
			"a_id": "name_3",
			"password": "1234"
		}
	]
}
```

안전한 코드일때
```json
{
	"data": [
		{
			"id": 1,
			"name": "이름_a",
			"a_id": "name_1",
			"password": "1234"
		}
	]
}
```
위와 같이 취약한 코드는 모든 row를 다 보여주고 안전한 코드는 의도한 대로 id = 1인 row만 보여준다.

